x/perp/amm/abci_test.go

```diff
@@ -31,21 +31,21 @@ func TestSnapshotUpdates(t *testing.T) {
 	ctx = ctx.WithBlockTime(time.Date(2015, 10, 21, 0, 0, 0, 0, time.UTC)).WithBlockHeight(1)
 
 	require.NoError(t, perpammKeeper.CreatePool(
-		ctx,
-		asset.Registry.Pair(denoms.BTC, denoms.NUSD),
-		sdk.NewDec(1_000),
-		sdk.NewDec(1_000),
-		*types.DefaultMarketConfig().
-			WithTradeLimitRatio(sdk.OneDec()).
-			WithFluctuationLimitRatio(sdk.OneDec()),
-		sdk.ZeroDec(),
-		sdk.OneDec(),
+		/* ctx */ ctx,
+		/* pair */ asset.Registry.Pair(denoms.BTC, denoms.NUSD),
+		/* quoteReserve */ sdk.NewDec(1_000),
+		/* baseReserve */ sdk.NewDec(1_000),
+		/* config */ *types.DefaultMarketConfig().WithTradeLimitRatio(sdk.OneDec()).WithFluctuationLimitRatio(sdk.OneDec()),
```

Comment:
I would set defaultmarketconfig as type, not pointer, to avoid this casting.

Reply:
Golang doesn't have the default args feature, you have have to use pointer if you want to see default values inside a constructor

Reply:
It should be possible to pass a struct instead of a pointer to a struct.